src/codemod.ts

```diff
@@ -0,0 +1,115 @@
+import { applyPatch } from 'diff'
+import * as sourcegraph from 'sourcegraph'
+import { Settings } from './settings'
+import { Change, Match, SourceTree } from './types'
+
+function sourcegraphURL(): URL {
+    const url =
+        sourcegraph.configuration.get<Settings>().value['pr.sourcegraphUrl'] ||
+        sourcegraph.internal.sourcegraphURL.toString()
+    try {
+        return new URL(url)
+    } catch (e) {
+        if ('message' in e && /Invalid URL/.test(e.message)) {
```

Comment:
The URL constructor can never throw of any other reason than an invalid URL, so this check is not necessary. In general, error messages should not be inspected programmatically.

Reply:
Good to know.